dvc/exceptions.py

```diff
@@ -118,8 +118,8 @@ class NotDvcRepoError(DvcException):
         root (str): path to the directory.
     """
 
-    def __init__(self, root):
-        msg = (
+    def __init__(self, root, msg=None):
+        msg = msg or (
```

Comment:
While we are at it, let's not include default message in the exception itself but rather provide the message when we raise. I think the only place where this is raised in `find_root`, so shouldn't be too hard to adjust, unless there are too many tests that depend on it.